go-mode.el

```diff
@@ -1156,11 +1157,14 @@ with goflymake \(see URL `https://github.com/dougm/goflymake'), gocode
           (setq our-gofmt-args (append our-gofmt-args
                                        gofmt-args
                                        (list "-w" tmpfile)))
-          (message "Calling gofmt: %s %s" gofmt-command our-gofmt-args)
+
+	  (setq gofmt-full-cmd (format "GOPATH=%s:%s %s %s" (go-guess-gopath) (getenv "GOPATH") gofmt-command (mapconcat 'identity our-gofmt-args " ")))
```

Comment:
This command format(+ `shell-command`) works only on `sh`, `bash`, `zsh`(`shell-command` uses `SHELL` environment variable shell for executing command). This raises error on `csh`, `tcsh`, `fish`(and Windows). I suppose that you should use `process-environment` variable for setting environment variable temporarily.

See
- https://www.gnu.org/software/emacs/manual/html_node/elisp/System-Environment.html